### PR TITLE
return correct ipaddress for openvz guests (fixes #415)

### LINKED ISSUE #415: ohai ipaddress returns loopback address on servers virtualized through OpenVZ
So what about OHAI-237 feature? In short: in openvz guest system ohai detect ip 127.0.0.2. not real.
More info and patch available in JIRA thread: https://tickets.opscode.com/browse/OHAI-237?focusedCommentId=49337&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-49337

### PR DESCRIPTION
**Pull request by MichaelSp**
I had to refactor the `linux/network.rb` to understand what's going on and how to fix the issue with the openvz-guest ipaddress. 
But finally I fixed it @ line 92

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @MichaelSp

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Comment by cmluciano** [comment]
Seems that the travis error may have been a fluke. The tests pass locally.

File: lib/ohai/plugins/linux/network.rb [hunk 1]
```diff
@@ -41,6 +41,246 @@ def iproute2_binary_available?
...
 
+  def is_openvz?
+    ::File.directory?('/proc/vz')
+  end
+
+  def is_openvz_host?
+    is_openvz? && ::File.directory?('/proc/bc')
+  end
+
+  def extract_neighbours(family, iface, neigh_attr)
```
Comment (mcquin) [inline lib/ohai/plugins/linux/network.rb]: Chef and chef projects are using US English. For consistency, could you use "neighbors"?

**Comment by mcquin** [comment]
Great refactor! :)

**Comment by mcquin** [comment]
@chef/client-core

**Comment by thommay** [comment]
:+1:

**Comment by lamont-granquist** [comment]
:+1:

**Pull request merged**